fuzzydir.lua

```diff
@@ -20,18 +20,38 @@
     Determines the max depth of recursive search, should be >= 1
 
     Examples for "sub-file-paths = **":
-    "max_search_depth = 1" => finds [xyz.ass, subs/xyz.ass]
-    "max_search_depth = 2" => finds [xyz.ass, subs/xyz.ass, subs/moresubs/xyz.ass]
+    "max_search_depth = 1" => finds [subs/xyz.ass]
+    "max_search_depth = 2" => finds [subs/xyz.ass, subs/moresubs/xyz.ass]
 
     Please be careful when setting this value too high as it can result in awful performance or even stack overflow
 ]]
-local max_search_depth = 3
 
-local utils = require "mp.utils"
+local msg = require 'mp.msg'
+local options = require 'mp.options'
+local utils = require 'mp.utils'
+
+o = {
+    max_search_depth = 3, --determines the max depth of recursive search, should be >= 1
+    excluded_dir = [[
+        ["?:"]
+        ]], --excluded directories for cloud mount disks on Windows, example: ["X:", "Z:"]. !!the option only for Windows
+}
+options.read_options(o)
+
+o.excluded_dir = utils.parse_json(o.excluded_dir)
 
 local default_audio_paths = mp.get_property_native("options/audio-file-paths")
 local default_sub_paths = mp.get_property_native("options/sub-file-paths")
 
+function starts_protocol(tab, val)
+	for index, value in ipairs(tab) do
+		if (val:find(value) == 1) then
```

Comment:
This is rare syntax, which most other languages don't have. Should be changed to `string.find(val, value)` which is more accessible for people not familiar with Lua IMO.